### ISSUE TITLE
[bug] The parameter add_special_tokens has no effect when set to False in Bert WordPiece tokenizer

### ISSUE THREAD

**Issue by kzimmer-bah** [huggingface/tokenizers#141]
The fix for issue #113 removes the utility of the `add_special_tokens` parameter. It forcibly adds the unk, sep, and cls tokens despite the value of this parameter.  The offending line: 

https://github.com/huggingface/tokenizers/blob/5e612669bbbad431a400c294b54b3ad6eb5e3f82/bindings/python/tokenizers/implementations/bert_wordpiece.py#L33

should be moved inside logic checking the status of the `add_special_tokens` parameter. I have a trained Bert WordPiece tokenizer that should not include these tokens but no way of loading it correctly.

**Comment by n1t0** [comment]
This is just a way of telling the tokenizer that these tokens should be treated as special tokens when seen in the input. It just means that the tokenizer won't try to split them. 

Is this a problem for you? I mean, have you encountered any unexpected behavior?

**Comment by kzimmer-bah** [comment]
It seems like they are being added to the tokenizer's vocabulary where they were not before. On version 0.2.1 I trained a model that included a custom '<UNK>' token but neither sep nor cls. I trained the tokenizer with `add_special_tokens=False`. The vocab size when loaded was 10000. 

In versions >0.2.1, the model has vocab size 10002 when loaded and adds index 10000 and 10001 for tokens "[SEP]" and "[CLS]" respectively. I only noticed because I was loading a pretrained embedding matrix with 10000 rows that crashed due to an index mismatch.
